src/port1.0/portextract.tcl

```diff
@@ -152,5 +152,17 @@ proc portextract::extract_main {args} {
 
         chownAsRoot ${extract.dir}
     }
+
+    # If expected path of extract doesn't exist && worksrcdir is
+    # not explicitly set to subdirectory, symlink to actual
+    # path.
+    if {![file isdirectory $worksrcpath] && [regexp {^[^/]+$} $worksrcdir]} {
+        set workdirs [glob -nocomplain -types d [file join $workpath "*"]]
+        if {[llength $workdirs] == 1} {
+            set dir [file tail [lindex $workdirs 0]]
+            ui_info "$UI_PREFIX  [format [msgcat::mc "Symlink: %s -> %s"] $worksrcpath $dir]"
```

Comment:
I think I would use `ui_debug` here instead of `ui_info`, and I would not use `$UI_PPREFIX`.